contracts/test/GrantRegistry.test.ts

```diff
@@ -0,0 +1,222 @@
+// --- External imports ---
+import { artifacts, ethers, waffle } from 'hardhat';
+import { BigNumber, BigNumberish } from 'ethers';
+import { Artifact } from 'hardhat/types';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { expect } from 'chai';
+
+// --- Our imports ---
+import { GrantRegistry } from '../typechain/GrantRegistry';
+
+// --- Parse and define helpers ---
+const { deployContract } = waffle;
+const { isAddress } = ethers.utils;
+const randomAddress = () => ethers.Wallet.createRandom().address;
+const randomPtr = () => String(Math.random()); // doesn't need to be a valid URL for testing
+const BN = (x: BigNumberish) => BigNumber.from(x);
+
+// --- Type definitions ---
+// The output from ethers/typechain allows array or object access to grant data, so we must define types for
+// handling the Grant struct as done below
+type GrantObject = {
+  id: BigNumber;
+  owner: string;
+  payee: string;
+  metaPtr: string;
+};
+type GrantArray = [BigNumber, string, string, string];
+type GrantEthers = GrantArray & GrantObject;
+type Grant = GrantObject | GrantEthers;
+
+// --- GrantRegistry tests ---
+describe('GrantRegistry', function () {
+  let deployer: SignerWithAddress, user1: SignerWithAddress, user2: SignerWithAddress;
+  let registry: GrantRegistry;
+
+  // Helper method to create a grant and return the data used to create it
+  async function createGrant(owner: string, payee: string, metaPtr: string, user?: SignerWithAddress) {
+    user = user ? user : deployer; // if a signer, `user`, is not specified we use the `deployer` account
+    await registry.connect(user).createGrant(owner, payee, metaPtr);
+    return { owner, payee, metaPtr };
+  }
+
+  // Helper method to verify that two Grant objects are equal
+  function expectEqualGrants(grant1: Grant, grant2: Grant) {
+    expect(grant1.id).to.equal(grant2.id);
+    expect(grant1.owner).to.equal(grant2.owner);
+    expect(grant1.payee).to.equal(grant2.payee);
+    expect(grant1.metaPtr).to.equal(grant2.metaPtr);
+  }
```

Comment:
might be nice to move these into a test util file as well

Reply:
Hm, where else would we need to check that two Grant objects are equal?

Reply:
Off the top of my head, one example might be a test of `GrantRounds` that shows the same grant can exist in two rounds or something? I think we'd want utilities like this in a generic place so that they're available and discoverable for future test authors who might be working on a new test, and wouldn't know to look for them in an existing test they didn't author.

Reply:
That makes sense, done in d263ee4b655acd56d3cc4634253fdf38f761febf